Documentation/api-reference.md

```diff
@@ -1,10 +1,302 @@
+                               _ completion: (() -> Void)? = nil)
+```
+
+##### Example
+```swift
+import AEPEdgeIdentity
+
+...
+MobileCore.registerExtensions([Identity.self])
+
+##### Syntax
+```objectivec
++ (void) registerExtensions: (NSArray<Class*>* _Nonnull) extensions
+                 completion: (void (^ _Nullable)(void)) completion;
+
+##### Example
+```objectivec
+@import AEPEdgeIdentity;
+
+...
+[AEPMobileCore registerExtensions:@[AEPMobileEdgeIdentity.class] completion:nil];
+```
+
+------
+
+### removeIdentity
+
+Remove the identity from the stored client-side [IdentityMap](#identitymap). The Identity extension will stop sending the identifier to the Edge Network. Using this API does not remove the identifier from the server-side User Profile Graph or Identity Graph.
+
+Identities with an empty _id_ or _namespace_ are not allowed and are ignored.
+
+Removing identities using a reserved namespace is not allowed using this API. The reserved namespaces are:
+
+* ECID
+* IDFA
+* GAID
+
+#### Swift
+
+##### Syntax
+```swift
+Identity.removeIdentity(item: IdentityItem(id: "user@example.com"), withNamespace: "Email")
+```
+
+##### Example
+```swift
+static func removeIdentity(item: IdentityItem, withNamespace: String)
+```
+
+#### Objective-C
+
+
+##### Syntax
+```objectivec
++ (void) removeIdentityItem:(AEPIdentityItem * _Nonnull) item
+                             withNamespace: (NSString * _Nonnull) namespace
+```
+
+##### Example
+```objectivec
+AEPIdentityItem *item = [[AEPIdentityItem alloc] initWithId:@"user@example.com" authenticatedState:AEPAuthenticatedStateAuthenticated primary:false];
+[AEPMobileEdgeIdentity removeIdentityItem:item withNamespace:@"Email"];
+```
+
+------
+
+### resetIdentities
+
+Clears all identities stored in the Identity extension and generates a new Experience Cloud ID (ECID). Using this API does not remove the identifiers from the server-side User Profile Graph or Identity Graph.
+
+This is a destructive action, since once an ECID is removed it cannot be reused. The new ECID generated by this API can increase metrics like unique visitors when a new user profile is created.
+
+Some example use cases for this API are:
+* During debugging, to see how new ECIDs (and other identifiers paired with it) behave with existing rules and metrics.
+* A last-resort reset for when an ECID should no longer be used.
+
+This API is not recommended for:
+* Resetting a user's consent and privacy settings; see [Privacy and GDPR](https://aep-sdks.gitbook.io/docs/resources/privacy-and-gdpr).
+* Removing existing custom identifiers; use the [`removeIdentity`](#removeidentity) API instead.
+* Removing a previously synced advertising identifier after the advertising tracking settings were changed by the user; use the [`setAdvertisingIdentifier`](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/identity/identity-api-reference#setadvertisingidentifier) API instead.
+
+> :warning: **Warning**
```

Comment:
We should use the same format for all info/warning blocks. If you like the icon format more 😃 , can you please update all occurrences?

Reply:
I noticed the warning itself has icon added by github, so removing my custom icon.